### PR TITLE
[IMP] purchase_requisition: rfq from blanket should set date_planned from seller delay

### PR DESCRIPTION
**Pull request by matt454357**
**Description of the issue/feature this PR addresses:**
A Purchase Agreement can be used as a blanket purchase order.  Blanket orders typically have long duration, e.g. 6 to 18 months.  Purchase Orders are created periodically to request a release against the blanket.  When a user creates a release order (purchase order), it should show a scheduled delivery date based on some prearranged seller delay.  In short, the Scheduled Date on the RFQ should default to today plus the seller delay.  It currently defaults to today.
    
**Current behavior before PR:**
* When purchase agreement type = Blanket Order
* When purchase agreement Delivery Date is null (which it should be for blanket type agreements)
* When a product.supplierinfo record exists for this res_partner and this product
* When creating a new rfq by clicking NEW QUOTATION
* Then the Scheduled Date for each PO line is currently set to today
    
**Desired behavior after PR is merged:**
* When purchase agreement type = Blanket Order
* When a product.supplierinfo record exists for this res_partner and this product
* When creating a new rfq by clicking NEW QUOTATION
* Then the Scheduled Date for each PO line will be calculated as today plus the seller delay

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/purchase_requisition/models/purchase_requisition.py [hunk 1]
```diff
@@ -160,17 +161,23 @@ def _onchange_product_id(self):
...
-    def _prepare_purchase_order_line(self, name, product_qty=0.0, price_unit=0.0, taxes_ids=False):
+    def _prepare_purchase_order_line(self, name, product_qty=0.0, price_unit=0.0, taxes_ids=False, partner_id=False, po=False):
         self.ensure_one()
         requisition = self.requisition_id
+        seller = self.product_id._select_seller(
+            partner_id=partner_id,
+            quantity=product_qty,
+            date=requisition.schedule_date or fields.Date.today(),
+            uom_id=self.product_id.uom_po_id)
+        date_planned = requisition.schedule_date or self.env['purchase.order.line']._get_date_planned(seller, po=po).strftime(DEFAULT_SERVER_DATETIME_FORMAT)
```
Comment (xmo-odoo) [inline addons/purchase_requisition/models/purchase_requisition.py]: `fields.Datetime` has helpers to convert between datetime object and the server format, there is no reason to invoke strftime manually.

Comment (matt454357) [reply addons/purchase_requisition/models/purchase_requisition.py]: Hi @xmo-odoo.  Here's a commit to fix that.

**Comment by matt454357** [comment]
Would it be better to add a field on the requisition line, for delivery lead time?  Improved delivery lead times is one of the benefits of having a blanket purchase agreement.  And, these would be different from the product supplier delivery lead time.

**Comment by C3POdoo** [comment]
Dear @matt454357,

Thank you for your report but we are closing it due to inactivity.
We apology if we could not look at your request in time.
If your report still makes sense, don't hesitate to reopen a new one. We will try to check it as soon as possible.

_This is an automated message._

**Pull request closed without merging**